src/rulesets/LinearAlgebra/dense.jl

```diff
@@ -394,3 +394,83 @@ function rrule(
     end
     return Ω, lyap_pullback
 end
+
+#####
+##### `kron`
+#####
+
+function frule((_, Δx, Δy), ::typeof(kron), x, y)
+    return kron(x, y), kron(Δx, y) + kron(x, Δy)
+end
+
+function rrule(::typeof(kron), x::AbstractVector, y::AbstractVector)
+    function kron_pullback(z̄)
+        x̄ = zero(x)
+        ȳ = zero(y)
+        m = firstindex(z̄)
+        @inbounds for i in eachindex(x)
+            xi = x[i]
+            for k in eachindex(y)
+                x̄[i] += y[k]' * z̄[m]
+                ȳ[k] += xi' * z̄[m]
```

Comment:
This won't work with immutable arrays. I wonder if the whole rule could be rewritten using `map`/`broadcast`/`dot` or some other linalg operations that do not require preallocating the output array.

Reply:
This is a really good point -- @simsurace we will need this to work with StaticArrays in order to get TemporalGPs working again

Reply:
It may be advantageous to do that, yes.